src/instructlab/model/train.py

```diff
@@ -521,20 +615,333 @@ def get_files(directory: str, pattern: str) -> list[str]:
-                f"Error while executing training library {exc}",
+                f"CHECKPOINT EVALUATION: {ckpt_path} SCORED {ckpt_score}",
                 fg="red",
+                bg="cyan",
             )
-            raise click.exceptions.Exit(1)
+            results.append((ckpt_score, ckpt_path))
+
+    if not results:
+        raise RuntimeError(
+            "_evaluate_dir_of_checkpoints - No checkpoints were evaluated successfully at %s. No scores were recored."
+            % checkpoints_dir_path
+        )
+
+    ckpt_performance: list[tuple[float, pathlib.Path]] = sorted(results, reverse=True)
+
+    # (score, path)
+    return ckpt_performance[0]
+
+
+def _run_phased_training(
+    ctx,
+    train_args: TrainingArgs,
+    torch_args: TorchrunArgs,
+    base_dir: pathlib.Path,
+    phase1_data: pathlib.Path,
+    phase1_num_epochs: int | None,
+    phase1_samples_per_save: int | None,
+    phase2_data: pathlib.Path,
+    phase2_num_epochs: int | None,
+    phase2_samples_per_save: int | None,
+    mtbench_judge: pathlib.Path,
+) -> None:
+    p1_ckpt_dir: pathlib.Path = base_dir / "phase1" / "checkpoints"
+    p2_ckpt_dir: pathlib.Path = base_dir / "phase2" / "checkpoints"
+
+    if not p1_ckpt_dir.is_dir() and p2_ckpt_dir.is_dir():
+        raise FileNotFoundError(
+            "Paths to checkpoint storage for phase1 and phase2 must be directories."
+        )
+
+    if not phase1_data.exists() and phase2_data.exists():
+        raise FileNotFoundError(
+            "Data for both phase1 and phase2 must be specified for phased training."
+        )
+
+    if not mtbench_judge.resolve().exists():
+        raise FileNotFoundError(
+            f"MT-Bench model direcory could not be found: {mtbench_judge}\nMust be an absolute path to a model directory."
+        )
+
+    # -----------------------------------------------------------------------------
+    click.secho("Training Phase 1/2...", fg="cyan")
+
+    p1_train_args: TrainingArgs = train_args.model_copy(deep=True)
+
+    if phase1_num_epochs:
+        logger.debug(f"Phase 1 epochs set to: {phase1_num_epochs}")
+        p1_train_args.num_epochs = phase1_num_epochs
+
+    if phase1_samples_per_save:
+        logger.debug(f"Phase 2 samples per save set to: {phase2_samples_per_save}")
+        p1_train_args.save_samples = phase1_samples_per_save
+
+    click.secho(
+        f"Phase 1/2 training arguments:\n{pprint.pformat(train_args)}", fg="cyan"
+    )
+
+    _training_phase(
+        train_args=p1_train_args,
+        torch_args=torch_args,
+        data_path=phase1_data,
+        ckpt_dir=p1_ckpt_dir,
+        # model override not necessary because we expect model to come from ctx.params.model_path.
+    )
+
+    # -----------------------------------------------------------------------------
+    click.secho("MMLU evaluation for Phase 1...", fg="cyan")
+    hf_p1_ckpts: pathlib.Path = p1_ckpt_dir / "hf_format"
+
+    _, p1_best_ckpt = _evaluate_dir_of_checkpoints(
```

Comment:
Should we lot the interim score maybe too?

Reply:
It's getting written in `_evaluate_dir_of_checkpoints`